modules/fluid_properties/include/userobjects/IdealGasFluidProperties.h

```diff
@@ -20,8 +20,9 @@ class IdealGasFluidProperties : public SinglePhaseFluidProperties
   virtual Real pressure(Real v, Real u) const;
   virtual Real temperature(Real v, Real u) const;
   virtual Real c(Real v, Real u) const;
-  virtual Real cp(Real v, Real u) const;
-  virtual Real cv(Real v, Real u) const;
+  virtual Real cp(Real v=0., Real u=0.) const;
+  virtual Real cv(Real v=0., Real u=0.) const;
```

Comment:
`cv` and `cp` are typically computed via `FluidPropertiesMaterial` which requires specific volume and internal energy, so I do not think this is really needed.

In cases where this would be called from thing like the constraint system, we still have to provide internal energy and specific volume, because the user can switch the fluid properties class in the input file. This is my experiance which using this in RELAP-7.

Did you run into a case where, internal energy and specific volume would not be needed?

Similarly for `gamma` below...